### ISSUE TITLE
Fuzzy search not working

### ISSUE THREAD

**Issue by Tobotimus** [Cog-Creators/Red-DiscordBot#2031]
# Other bugs

It appears we have an issue with fuzzy search. When enabled in DMs, sending a misspelled command yields this:
```
[19/08/2018 19:47] ERROR events on_error 181: Exception in on_command_error
Traceback (most recent call last):
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/.venv/src/discord.py/discord/client.py", line 220, in _run_event
    await coro(*args, **kwargs)
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/events.py", line 234, in on_command_error
    fuzzy_result = await fuzzy_command_search(ctx, ctx.invoked_with)
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/utils/__init__.py", line 52, in fuzzy_command_search
    is_alias, alias = await alias_cog.is_alias(ctx.guild, term)
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/cogs/alias/alias.py", line 60, in is_alias
    server_aliases = await self.unloaded_aliases(guild)
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/cogs/alias/alias.py", line 41, in unloaded_aliases
    return (AliasEntry.from_json(d) for d in (await self._aliases.guild(guild).entries()))
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/config.py", line 761, in guild
    return self._get_base_group(self.GUILD, guild.id)
AttributeError: 'NoneType' object has no attribute 'id'
```

~~And server fuzzy doesn't seem to work at all~~ (Edit: It was just a high ratio, I'm going to lower it). I'll investigate.

**Comment by Tobotimus** [comment]
Having resolved the above traceback on my local branch, I ran into this one which seems to be caused by typing a command similar to a guild-only one whilst in DM:
```
Traceback (most recent call last):
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/.venv/src/discord.py/discord/client.py", line 220, in _run_event
    await coro(*args, **kwargs)
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/events.py", line 231, in on_command_error
    fuzzy_result = await fuzzy_command_search(ctx, ctx.invoked_with)
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/utils/__init__.py", line 69, in fuzzy_command_search
    ctx, extracted
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/utils/__init__.py", line 33, in filter_commands
    for i in extracted
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/utils/__init__.py", line 37, in <listcomp>
    and all([await p.can_run(ctx) for p in i[0].parents])
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/redbot/core/utils/__init__.py", line 37, in <listcomp>
    and all([await p.can_run(ctx) for p in i[0].parents])
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/.venv/src/discord.py/discord/ext/commands/core.py", line 734, in can_run
    return (await discord.utils.async_all(predicate(ctx) for predicate in predicates))
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/.venv/src/discord.py/discord/utils.py", line 276, in async_all
    for elem in gen:
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/.venv/src/discord.py/discord/ext/commands/core.py", line 734, in <genexpr>
    return (await discord.utils.async_all(predicate(ctx) for predicate in predicates))
  File "/home/tobyh/PycharmProjects/Red-DiscordBot/.venv/src/discord.py/discord/ext/commands/core.py", line 1282, in predicate
    raise NoPrivateMessage('This command cannot be used in private messages.')
discord.ext.commands.errors.NoPrivateMessage: This command cannot be used in private messages.
```

**Comment by Tobotimus** [comment]
Resolved by #2122.
